prboom2/src/d_main.c

```diff
@@ -1510,6 +1512,29 @@ static void EvaluateDoomVerStr(void)
   lprintf(LO_INFO, "Playing: %s\n", doomverstr);
 }
 
+static void D_AddZip(const char* zipped_file_name)
+{
+  dsda_string_t temporary_directory = { NULL, 0 };
+  char* full_zip_path = I_FindZip(zipped_file_name);
```

Comment:
Please use the aforementioned `I_RequireZip` here. It's dsda-doom policy to abort if a requested file doesn't exist (and then you won't need the null check).